_plugins/tag_page_generator.rb

```diff
@@ -0,0 +1,39 @@
+module TagPageGeneratorPlugin
+    class TagPageGenerator < Jekyll::Generator
+      safe true
+  
+      def generate(site)
+        site.tags.each do |tag, posts|
+          site.pages << TagPage.new(site, tag, posts)
+        end
+      end
+    end
+  
+    # Subclass of `Jekyll::Page` with custom method definitions.
+    class TagPage < Jekyll::Page
+      def initialize(site, tag, posts)
+        @site = site             # the current site instance.
+        @base = site.source      # path to the source directory.
+        @dir  = "tagged"         # the directory the page will reside in.
```

Comment:
Does this need to change to "topic" so that the URL says "/topic/{tag}" instead of "/tagged/{tag}"?